### PR TITLE
Issue 20

### PR DESCRIPTION
**Pull request by rmroot**
connects #20 

Additional error handling can be added for v2

Isn't preventing same dates from same worksheets but will flag them in the tables once they are in

### CODE REVIEW

File: src/app/utility/energy-consumption/energy-consumption.component.html [hunk 1]
```diff
@@ -1,38 +1,38 @@
...
                 <!-- <span [hidden]="hasWasteWaterData">*</span> -->
             </a>
         </li>
-        <li [hidden]="!otherUtility" [class.red]="!hasOtherUtilityData">
-            <a routerLink="other-utility" [routerLinkActive]="['active']">Other Utility 
+        <li [hidden]="!otherUtility">
+            <a routerLink="other-utility" [routerLinkActive]="['active']" [class.red]="hasOtherUtilityError">Other Utility 
                 <!-- <span [hidden]="hasOtherUtilityData">*</span> -->
             </a>
         </li>
```
Comment (cranfillkd) [inline src/app/utility/energy-consumption/energy-consumption.component.html:38]: This whole page is incorrect. I recently fixed this in a PR.  [class.red]="xyz" should be on the list element. Not the anchor.

Screen Shot 2021-04-20 at 9 48 20 AM

File: src/app/utility/energy-consumption/energy-consumption.component.ts [hunk 2]
```diff
@@ -61,31 +63,47 @@ export class EnergyConsumptionComponent implements OnInit {
...
     });
   }
 
+  checkHasErrors(meter: IdbUtilityMeter, facilityMeterData: Array<IdbUtilityMeterData>) {
+    let meterData: Array<IdbUtilityMeterData> = facilityMeterData.filter(data => { return data.meterId == meter.id });
+    let dataHasErrors: Date = this.utilityMeterDataService.checkForErrors(meterData);
+    if(dataHasErrors){
+      return true;
+    }
+    return meterData.length == 0;
```
Comment (cranfillkd) [inline src/app/utility/energy-consumption/energy-consumption.component.ts:106]: Why return meterData.length == 0? Shouldn't it be false if no errors?

File: src/app/utility/energy-consumption/utility-meter-data/electricity-data-table/electricity-data-table.component.ts [hunk 3]
```diff
@@ -104,4 +105,15 @@ export class ElectricityDataTableComponent implements OnInit {
       this.orderDataField = str;
     }
   }
+
+  checkError(readDate: Date): boolean {
+    if (this.meterListItem.errorDate) {
+      let readDateItem: Date = new Date(readDate);
+      if (readDateItem.getUTCFullYear() == this.meterListItem.errorDate.getUTCFullYear() && readDateItem.getUTCMonth() && this.meterListItem.errorDate.getUTCMonth()) {
+        console.log('TRUE!')
```
Comment (cranfillkd) [inline src/app/utility/energy-consumption/utility-meter-data/electricity-data-table/electricity-data-table.component.ts:113]: Remove console log

File: src/app/utility/energy-consumption/utility-meter-data/general-utility-data-table/general-utility-data-table.component.ts [hunk 4]
```diff
@@ -99,4 +100,15 @@ export class GeneralUtilityDataTableComponent implements OnInit {
       this.orderDataField = str;
     }
   }
+
+  checkError(readDate: Date): boolean {
+    if (this.meterListItem.errorDate) {
+      let readDateItem: Date = new Date(readDate);
+      if (readDateItem.getUTCFullYear() == this.meterListItem.errorDate.getUTCFullYear() && readDateItem.getUTCMonth() == this.meterListItem.errorDate.getUTCMonth()) {
+        console.log('TRUE!')
```
Comment (cranfillkd) [inline src/app/utility/energy-consumption/utility-meter-data/general-utility-data-table/general-utility-data-table.component.ts:108]: remove console.log

File: src/app/utility/energy-consumption/utility-meter-data/electricity-data-table/electricity-data-table.component.css [hunk 5]
```diff
@@ -5,4 +5,8 @@ thead th.active {
 thead th:hover{
     cursor: pointer;
     text-decoration: underline;
+}
+
+td.error{
+    background: red !important;
```
Comment (cranfillkd) [inline src/app/utility/energy-consumption/utility-meter-data/electricity-data-table/electricity-data-table.component.css:11]: Might as well put this in global.css

File: src/app/utility/energy-consumption/utility-meter-data/general-utility-data-table/general-utility-data-table.component.css [hunk 6]
```diff
@@ -5,4 +5,7 @@ thead th.active {
 thead th:hover{
     cursor: pointer;
     text-decoration: underline;
+}
+td.error{
+    background: red !important;
```
Comment (cranfillkd) [inline src/app/utility/energy-consumption/utility-meter-data/general-utility-data-table/general-utility-data-table.component.css:10]: Global.css

File: src/app/utility/energy-consumption/utility-meter-data/utility-meter-data.component.ts [hunk 7]
```diff
@@ -223,4 +226,9 @@ export class UtilityMeterDataComponent implements OnInit {
   cancelBulkDelete() {
     this.showBulkDelete = false;
   }
+
+
+  checkDuplicateMonthsEntries() {
+
+  }
```
Comment (cranfillkd) [inline src/app/utility/energy-consumption/utility-meter-data/utility-meter-data.component.ts:233]: Empty function

**Review by cranfillkd — CHANGES_REQUESTED** [review]
feedback on the files

File: src/app/utility/energy-consumption/utility-meter-data/electricity-data-table/electricity-data-table.component.ts (hunk 3 above)
Comment (rmroot) [reply src/app/utility/energy-consumption/utility-meter-data/electricity-data-table/electricity-data-table.component.ts]: fixed

File: src/app/utility/energy-consumption/utility-meter-data/electricity-data-table/electricity-data-table.component.css (hunk 5 above)
Comment (rmroot) [reply src/app/utility/energy-consumption/utility-meter-data/electricity-data-table/electricity-data-table.component.css:11]: unused, leftover from testing. removed

File: src/app/utility/energy-consumption/utility-meter-data/general-utility-data-table/general-utility-data-table.component.css (hunk 6 above)
Comment (rmroot) [reply src/app/utility/energy-consumption/utility-meter-data/general-utility-data-table/general-utility-data-table.component.css:10]: unusued. leftover. removed

File: src/app/utility/energy-consumption/utility-meter-data/utility-meter-data.component.ts (hunk 7 above)
Comment (rmroot) [reply src/app/utility/energy-consumption/utility-meter-data/utility-meter-data.component.ts]: removed

File: src/app/utility/energy-consumption/energy-consumption.component.html (hunk 1 above)
Comment (rmroot) [reply src/app/utility/energy-consumption/energy-consumption.component.html:38]: Merging error. Wondering why it was yelling at me. My bad. Fixed

File: src/app/utility/energy-consumption/energy-consumption.component.ts (hunk 2 above)
Comment (rmroot) [reply src/app/utility/energy-consumption/energy-consumption.component.ts:106]: It was setup to display an error if a meter had no data. That's what this check is doing

Errors (for now):
1. duplicate month entries
2. meter has no data

File: src/app/utility/energy-consumption/utility-meter-data/general-utility-data-table/general-utility-data-table.component.ts (hunk 4 above)
Comment (rmroot) [reply src/app/utility/energy-consumption/utility-meter-data/general-utility-data-table/general-utility-data-table.component.ts]: unused function. Removed

**Review by cranfillkd — APPROVED** [review]

**Pull request merged**